xds/src/main/java/io/grpc/xds/XdsServerWrapper.java

```diff
@@ -432,15 +448,28 @@ public void onChanged(final LdsUpdate update) {
       }
     }
 
+    private boolean ipAddressesMatch(String ldsAddress) {
+      HostAndPort ldsAddressHnP = HostAndPort.fromString(ldsAddress);
+      HostAndPort listenerAddressHnP = HostAndPort.fromString(listenerAddress);
+
+      InetAddress listenerIp = InetAddresses.forString(listenerAddressHnP.getHost());
+      InetAddress ldsIp = InetAddresses.forString(ldsAddressHnP.getHost());
+
+      if (listenerIp.isAnyLocalAddress()) {
+        return true;
+      }
+      return listenerIp.equals(ldsIp) && ldsAddressHnP.getPort() == listenerAddressHnP.getPort();
+    }
+
     @Override
     public void onResourceDoesNotExist(final String resourceName) {
       if (stopped) {
         return;
       }
       StatusException statusException = Status.UNAVAILABLE.withDescription(
-          String.format("Listener %s unavailable, xDS node ID: %s", resourceName,
+          String.format("%s listener unavailable, xDS node ID: %s", resourceName,
```

Comment:
nit: what would be the reason to switch the order of the error format here? I think `Listener %s` is slightly more common in the code base. Consistent formatting helps with searching when debugging issues.